tests/TestCase/Controller/ControllerTest.php

```diff
@@ -626,7 +626,7 @@ public function testRefererSlash()
 
         $controller = new Controller($request);
         $result = $controller->referer('/some/path', true);
-        $this->assertEquals('/base/some/path', $result);
+        $this->assertEquals('/some/path', $result);
```

Comment:
This issue only occurs when running the app in a subfolder which would be pretty rare in production. Plus I believe the major usage of getting referrer is to use it for redirection so anyone having their app in subfolder would be removing removing the base himself for the redirection to work properly. So they probability of affecting existing app is very low. That said I am fine with moving the patch to 3.next.